packages/kbn-alerts-ui-shared/src/common/hooks/use_alerts_data_view.ts

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useEffect, useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import { DataViewsContract, FieldSpec } from '@kbn/data-views-plugin/common';
+import { AlertConsumers, ValidFeatureId } from '@kbn/rule-data-utils';
+import type { ToastsStart, HttpStart } from '@kbn/core/public';
+
+import { DataViewBase } from '@kbn/es-query';
+import { useVirtualDataViewQuery } from './use_virtual_data_view_query';
+import { useFetchAlertsFieldsQuery } from './use_fetch_alerts_fields_query';
+import { useFetchAlertsIndexNamesQuery } from './use_fetch_alerts_index_names_query';
+
+export interface UseAlertsDataViewParams {
+  // Dependencies
+  http: HttpStart;
+  dataViewsService: DataViewsContract;
+  toasts: ToastsStart;
+
+  // Params
+  /**
+   * Array of feature ids used for authorization and area-based filtering
+   *
+   * Security data views must be requested in isolation (i.e. `['siem']`). If mixed with
+   * other feature ids, the resulting data view will be empty.
+   */
+  featureIds: ValidFeatureId[];
+}
+
+export interface UseAlertsDataViewResult {
+  isLoading: boolean;
+  dataView?: Omit<DataViewBase, 'fields'> & { fields: FieldSpec[] };
+}
+
+/**
+ * Computes a {@link DataViewBase} object for alerts indices based on the provided feature ids
+ *
+ * @returns
+ * A {@link DataViewBase} object, intentionally not typed as a complete {@link DataView} object
+ * since only Security Solution uses an actual in-memory data view (when `featureIds = ['siem']).
+ * In all other cases the data view is computed from the index names and fields fetched from the
+ * alerting APIs.
+ */
+export const useAlertsDataView = ({
+  http,
+  dataViewsService,
+  toasts,
+  featureIds,
+}: UseAlertsDataViewParams): UseAlertsDataViewResult => {
+  const includesSecurity = featureIds.includes(AlertConsumers.SIEM);
```

Comment:
I find these conditions a bit more eloquent than the [previous ones](https://github.com/elastic/kibana/pull/189638/files#diff-8e7de7345f3f13823c76709ff7d5a53f65664d2a12dcea8fa686291282457b4fL31)